bokehjs/src/lib/models/widgets/numeric_input.ts

```diff
@@ -69,14 +68,17 @@ export class NumericInputView extends InputWidgetView {
   render(): void {
     super.render()
 
-    this.input_el = input({
-      type: "text",
-      class: bk_input,
-      name: this.model.name,
-      value: this.format_value,
-      disabled: this.model.disabled,
-      placeholder: this.model.placeholder,
-    })
+    this.input_el = Object.assign(
+      input({
+        type: "text",
+        class: bk_input,
+        name: this.model.name,
+        value: this.format_value,
+        disabled: this.model.disabled,
+        placeholder: this.model.placeholder,
+      }),
+      {oldValue: this.format_value}
```

Comment:
I would avoid this kind of code, and just set `old_value` on `this` and incorporate `setInputFilter` into the class.